app/src/main/java/com/willowtree/vocable/CategoriesUseCase.kt

```diff
@@ -75,13 +85,15 @@ class CategoriesUseCase(
         storedCategoriesRepository.updateCategorySortOrders(categorySortOrders)
     }
 
-    override suspend fun addCategory(categoryName: String, sortOrder: Int) {
+    override suspend fun addCategory(categoryName: String) {
+        val allCategories = categories().first()
+
         storedCategoriesRepository.upsertCategory(
             Category.StoredCategory(
                 uuidProvider.randomUUIDString(),
                 LocalesWithText(mapOf(Pair(localeProvider.getDefaultLocaleString(), categoryName))),
                 false,
-                sortOrder
+                allCategories.size
             )
         )
     }
```

Comment:
theoretically if we have 7 presets

```
sortOrders: [0, 1, 2, 3, 4, 5, 6]
```

add one category

```
sortOrders: [0, 1, 2, 3, 4, 5, 6, 7]
```

then delete a preset:
```
sortOrders: [1, 2, 3, 4, 5, 6, 7]
```

if we wanted to add another category then we end up with two categories with the same sortOrder:

```
sortOrders: [1, 2, 3, 4, 5, 6, 7, 7]
```

Seems like it could cause (very minor) problems/non-determinism. I don't forsee any big problems but maybe we could avoid the problem altogether by doing something like `categories.maxBy { it.sortOrder } + 1` instead?

Reply:
Yea that's a really good catch. Trying to thread the needle of not needing logic that maintains perfectly continuous numbers, but also not breaking things

Reply:
There are also definitely issues with re-sorting in the app right now. I'm going to merge and continue moving forward so that this logic is at least out of the add category ViewModel, and follow up to fix things in their new home.